includes/process-purchase.php

```diff
@@ -74,6 +74,30 @@ function edd_process_purchase_form() {
 		'address'    => $user['address']
 	);
 
+	// Update a customer record if they have added/updated information
+	$customer = new EDD_Customer( $user_info['user_id'], true );
```

Comment:
This should be instantiated by email, not user ID, so that it works for guest purchases.